sklearn/preprocessing/label.py

```diff
@@ -53,6 +54,140 @@ def _check_numpy_unicode_bug(labels):
                            " NumPy to use LabelEncoder with unicode inputs.")
 
 
+class TargetTransformer(BaseEstimator):
+    """Meta-estimator to apply a transformation to the target before fitting
+
+    Useful for applying a non-linear transformation like np.log
+    to the target in a regression problem.
+
+    The computation during ``fit`` is::
+
+        estimator.fit(X, func(y))
+
+    The computation during ``predict`` is::
+
+        inverse_func(estimator.predict(X))
+
+
+    Parameter
+    ---------
+    estimator : object
+        Estimator object to wrap.
+
+    func : function
```

Comment:
Yeah I completely agree, I misread the first bullet earlier re:training portion. 

Come to think of it, [centering / standardization in linear models](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/base.py#L514) is also a good example of something that could be expressed under this form, and also confirms @amueller's point that we care about the score after inverse-transforming.